source/Connection/AMDSClientTCPSocket.cpp

```diff
@@ -19,6 +19,11 @@ AMDSClientTCPSocket::AMDSClientTCPSocket(const QString host, const quint16 port,
 	socketKey_ = "";
 	tcpSocket_ = new QTcpSocket(this);
 
+	waitingMorePackages_ = false;
+	readedBufferSize_ = 0;
+	expectedBufferSize_ = 0;
+	incomeDataBuffer_ = new QByteArray();
```

Comment:
How is it handled by Qt?  You are adding the byte array to the heap and by deleting the class, you are orphaning that pointer which will only get cleaned up when the program is closed.  That is definitely a memory leak.